### PR TITLE
Tests: improve test coverage for pymovement.utils.checks to account for error messages

### PR DESCRIPTION
**Pull request by SiQube**
while writing tests for `pymovements.utils.paths` I've realized that we have specific error message. I think we should test those within our test coverage.

### CODE REVIEW

**Review by dkrako — CHANGES_REQUESTED** [review]
Change requests:
- [ ] Just some minor styling for issues for the first three test cases

Thank you for this work!

I see that if every time just a ValueError is being checked to be raised, it's not very clear which one was actually raised.

I personally think that checking these messages usually isn't that necessary, but I don't mind having this.

**Comment by SiQube** [comment]
I also thought so -- but I found that in transforms we raise several TypeErrors (I think) and wanted to rather make sure that we actually raise the one where it is supposed to be. I'll talk to @theDebbister tomorrow about it

File: tests/utils/test_checks.py [hunk 1]
```diff
@@ -29,79 +29,98 @@
...
+        ),
+        pytest.param([1, 2, 3], None, '', id='non_zero_list_raises_no_error'),
+        pytest.param(
+            [1, 0, 3], ValueError,
+            'each component in variable must not be zero', id='zero_list_raises_value_error',
+        ),
+        pytest.param(np.array([1, 2, 3]), None, '', id='non_zero_np_array_raises_no_error'),
+        pytest.param(
+            np.array([1, 0, 3]), ValueError,
+            'each component in variable must not be zero', id='zero_np_array_raises_value_error',
```
[the remark this answers is not in the document]
Comment (SiQube) [reply tests/utils/test_checks.py:50]: done

File: tests/utils/test_checks.py [hunk 2]
```diff
@@ -29,79 +29,98 @@
...
-        pytest.param(np.array([1, 0, 3]), ValueError, id='zero_np_array_raises_value_error'),
+        pytest.param(5, None, '', id='non_zero_single_variable_raises_no_error'),
+        pytest.param(
+            0, ValueError,
+            'variable must not be zero', id='zero_single_variable_raises_value_error',
+        ),
+        pytest.param([1, 2, 3], None, '', id='non_zero_list_raises_no_error'),
+        pytest.param(
+            [1, 0, 3], ValueError,
+            'each component in variable must not be zero', id='zero_list_raises_value_error',
```
[the remark this answers is not in the document]
Comment (SiQube) [reply tests/utils/test_checks.py:44]: done.

File: tests/utils/test_checks.py [hunk 3]
```diff
@@ -29,79 +29,98 @@
...
-        pytest.param(5, None, id='non_zero_single_variable_raises_no_error'),
-        pytest.param(0, ValueError, id='zero_single_variable_raises_value_error'),
-        pytest.param([1, 2, 3], None, id='non_zero_list_raises_no_error'),
-        pytest.param([1, 0, 3], ValueError, id='zero_list_raises_value_error'),
-        pytest.param(np.array([1, 2, 3]), None, id='non_zero_np_array_raises_no_error'),
-        pytest.param(np.array([1, 0, 3]), ValueError, id='zero_np_array_raises_value_error'),
+        pytest.param(5, None, '', id='non_zero_single_variable_raises_no_error'),
+        pytest.param(
+            0, ValueError,
+            'variable must not be zero', id='zero_single_variable_raises_value_error',
```
[the remark this answers is not in the document]
Comment (SiQube) [reply tests/utils/test_checks.py:38]: done

**Comment by dkrako** [comment]
> I also thought so -- but I found that in transforms we raise several TypeErrors (I think) and wanted to rather make sure that we actually raise the one where it is supposed to be. I'll talk to @theDebbister tomorrow about it

Yes I see. I think our first priority is to reach 100% test coverage. After that, we can discuss more thorough testing in places where we may not be sure if the function does what it is expected to do.
Although 100% test coverage sounds great, it is no actual guarantee that every user scenario has been tested thoroughly (we don't even manage these).

For example in the tests for `Dataset.save()` I have only created tests that first completely remove the destination directory and then look if the correct directory was created. There is no actual checking if the created files match the internal dataframes. But the method gets 100% coverage so the task is finished regarding the minimal acceptance criteria.
I think this also holds true for the `utils.paths` tests, where we could add some more tests for some possible edge cases, but in the end it's a tradeoff between development productivity and risk reduction.

My personal way of thinking about this is, that I don't care much as long someone reports a problem with something.
Then an additional test has to be written that reproduces the problem and fails with the expected output.
But of course this approach is kind of problematic. For new features I try to adhere to test-driven-development and write tests before writing a line of code but with approaching deadlines my discipline is continuously declining..
So don't hesitate creating issues for this, beginner developers can then be pointed to these as these are usually easy to handle. Our own resources should be rather spent on more important stuff I think.

File: tests/utils/test_checks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkrako) [reply tests/utils/test_checks.py:38]: haha I actually meant that the lines with `id=...` should be on a separate line.
I don't mind if `0, ValueError, 'variable must not be zero,` is on a single line, that's left to your preference

**Review by dkrako — CHANGES_REQUESTED** [review]
Oh I think there was a misunderstanding regarding the requested style changes.

**Comment by SiQube** [comment]
force-push so no one saw 👍

**Review by dkrako — APPROVED** [review]
I clicked the wrong button LOL
